src/app/article/components/article/article.component.ts

```diff
@@ -1,10 +1,46 @@
-import { Component } from '@angular/core';
+import { Component, ElementRef, inject, OnInit, ViewChild } from '@angular/core';
+import { Router } from '@angular/router';
+import { Subject, takeUntil } from 'rxjs';
+import { SharedModalComponent } from 'src/app/shared/components/modal/shared-modal.component';
+import { IArticle } from '../../interfaces/article.interface';
+import { ArticlesService } from '../../services/articles.service';
 
 @Component({
   selector: 'app-article',
   templateUrl: './article.component.html',
   styleUrls: ['./article.component.scss']
 })
-export class ArticleComponent {
+export class ArticleComponent implements OnInit {
 
+  public articles: IArticle[];
+  public article: IArticle;
+  private articlesSvc = inject(ArticlesService);
+  private destroy$ = new Subject();
+  private router = inject(Router);
+  @ViewChild('modal', {static: false}) modalComponent: SharedModalComponent;
```

Comment:
`@ViewChild` can have the component class as a selector instead of the string with template variable name
`{static: false}` means that this template is inside of some condition, `*ngIf`, `*ngFor` ngSwitch ... and will not be in the dom from the start, no need to mark the template as `{static: false}` if the template is available from the start

```js
@ViewChild(SharedModalComponent') modalComponent: SharedModalComponent;
```